ql/src/java/org/apache/hadoop/hive/ql/processors/CompileProcessor.java

```diff
@@ -241,6 +255,14 @@ CommandProcessorResponse compile(SessionState ss) throws CommandProcessorExcepti
         out.closeArchiveEntry();
       }
       out.finish();
+      try {
+        Set<PosixFilePermission> perms = EnumSet.of(
+              PosixFilePermission.OWNER_READ,
+              PosixFilePermission.OWNER_WRITE);
+        Files.setPosixFilePermissions(Paths.get(testArchive.toURI()), perms);
+      } catch (IOException ioe) {
+        LOG.warn("Lockdown permissions could not be set for the jar archive. JAR file could be open to other users depending on default FS permissions");
```

Comment:
if there is an exception here, I am marking the file to be deleted when the JVM exists.
testArchive.deleteOnExit();

I assumed this would be used from its location in case of CLI fat client. So we cannot delete the jar until the CLI exits.